codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,81 @@
+import os
+import re
+import sys
+
+# Allow stdin, stdout, stderr, and the file descriptor for dynamic analysis log to remain open at exit
+ACCEPTABLE_OPEN_FDS = 4
+ERROR_EXIT_CODE = 123
+
+analysis_file_location = "../../build/Testing/Temporary"
```

Comment:
>Consider making this an argument

Do you mean as a command line argument?

>allow us to pass full directory path as an argument in the buildspec.


Passing the full directory path is difficult in our docker images. There is a series of number followed by src in the path which is not predictable. The path looks like this for example:
```
/codebuild/output/src3818542256/src/build/Testing/Temporary
```
We don't know what will replace `3818542256`.

Reply:
Given the above, I'm okay with the relative path but we should document
1. why this needs to be relative
2. what the associated directories are (e.g. who creates `Testing/Temporary`)
3. what directory this file needs to be run from.

On item 3, would it make more sense for us to expect this file to be run from the main directory?
```
python3 codebuild/bin/check_open_fds.py
```

Reply:
We don't need to use relative path, since I have found this codebuild environment [variable `CODEBUILD_SRC_DIR`](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html#:~:text=artifacts%20sample.-,CODEBUILD_SRC_DIR,-The%20directory%20path).

I have implemented it to make it take absolute path like this
https://github.com/boquan-fang/s2n-tls/blob/39456d8d4355ce89c4960cfa622107aa1c69ed41/codebuild/spec/buildspec_valgrind.yml#L74
The [test behaviors](https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/Valgrind/batch/Valgrind%3A2e161ecb-d54b-4ec9-b472-aec346459b21?region=us-west-2) didn't change, so I will do it the way @dougch has suggested.